loopchain/peer/block_manager.py

```diff
@@ -809,10 +828,13 @@ def verify_confirm_info(self, unconfirmed_block: Block):
             block_verifier.verify(unconfirmed_block,
                                   self.blockchain.last_block,
                                   self.blockchain,
-                                  ExternalAddress.fromhex_address(
-                                      self.blockchain.get_expected_generator(unconfirmed_block.header.peer_id)
-                                  ),
+                                  self.blockchain.get_expected_generator(unconfirmed_block.header.peer_id),
                                   reps_getter=reps_getter)
+        except NotInReps as e:
+            util.logger.debug(f"in _vote Not In Reps({e}) state({self.__channel_service.state_machine.state})")
+            self.__channel_service.switch_role()
+            self.__channel_service.broadcast_scheduler.update_audience(
+                self.__channel_service.peer_manager.prepared_reps_hash)
```

Comment:
This function doesn't stop at until `finally` block is called.
So please check finally block if it has no side effects (if you intended to quit the function at this line.